File: lecture/dingcorithm/2st_week/02_02_append_linked_list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


node = Node(3)
print(node.data, node.next)


class LinkedList:
    def __init__(self, value):
        self.head = Node(value)

# LinkedList의 가장 끝에 있는 노드에 새로운 노드를 연결해줘
    def append(self, value):
        cur = self.head

        while cur.next is not None:
            cur = cur.next

        cur.next = Node(value)

# LinkedList에서 저장한 head를 따라가면서 현재 있는 노드들을 전부 출력하는 함수
    def print_all(self):
        cur = self.head

        while cur is not None:
            print(cur.data)
            cur = cur.next

File: study/dingcorithm/2st_week/02_02_append_linked_list.js
class Node {
  constructor(data) {
    this.data = data;
    this.next = null;
  }
}

class LinkedList {
  constructor(data) {
    this.head = new Node(data);
  }

  append(data) {
    const newNode = new Node(data);

    // 빈 리스트일 경우도 고려필요
    if (!this.head) {
      this.head = newNode;
    }

    let cur = this.head;

    while (cur.next !== null) {
      cur = cur.next;
    }

    cur.next = newNode;
  }

  printAll() {
    let cur = this.head;

    while (cur !== null) {
      console.log(cur.data);
      cur = cur.next;
    }
  }
}

const sampleNode = new Node(3);
console.log(sampleNode);

const list = new LinkedList(3);
list.append(6);
list.append(7);
list.append(12);

list.printAll();
